specify/specify7
# Handle case when attachment relationship name is nonstandard

Fixes #4413 

The problem is that the following tables have a relationship to their attachment tables which is simply called `attachments` and not the 'expected' `tableNameAttachments`:
- DNASequence
- FieldNotebook
- FieldNotebookPage
- FieldNotebookPageSet

@realVinayak 
There are likely some places in `BulkAttachmentUpload` which could utilize this. 
Most notably: https://github.com/specify/specify7/blob/8907168c9785cebe8e82e6b787787412f95115de/specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/Upload.tsx#L396-L397
https://github.com/specify/specify7/blob/8907168c9785cebe8e82e6b787787412f95115de/specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/Upload.tsx#L434-L437
https://github.com/specify/specify7/blob/8907168c9785cebe8e82e6b787787412f95115de/specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/utils.ts#L332
While the affected tables are not currently supported for bulk uploading attachments with, I can imagine the tool be extended in the future to allow them. If this is likely, could this util be used in those places?


### Checklist

      and self-explanatory (or properly documented)

### Testing instructions
![Screenshot from 2024-01-16 21-12-41](https://github.com/specify/specify7/assets/64045831/1cb36b4d-c37c-44c0-bc1e-f0578aba779f)

For each of the following tables, ensure no error is thrown and attachments can be viewed when the `attachments` button is clicked (in the context of a viewing a single resource and in a recordset)
- DNASequence
- FieldNotebook
- FieldNotebookPage
- FieldNotebookPageSet

Additionally, ensure that the component still functions correctly for other tables with standard names for attachment relationships (CollectionObject, CollectingEvent, etc)
